src/Common/Commands.Common/AzurePSCmdlet.cs

```diff
@@ -137,6 +137,39 @@ protected string PSVersion
 
         protected abstract string DataCollectionWarning { get; }
 
+        private SessionState _asJobSessionState;
+
+        public SessionState AsJobSessionState
+        {
+            get
+            {
+                if (_asJobSessionState == null)
```

Comment:
I don't think we can instantiate this lazily, because it might be first referenced inside a job - I think we need to instantiate it either during BeginProcessing. or in the cmdlet constructor

Reply:
Done